registrations/models.py

```diff
@@ -636,6 +767,59 @@ def get_registration_message(self, subject, cleaned_body, plain_text_body):
             [self.email],
         )
 
+    def _get_confirmation_template_texts(
+        self, notification_type: SignUpNotificationType
+    ):
+        with (translation.override(self.get_service_language_pk())):
+            confirmation_message = self.registration.confirmation_message
+            event_name = self.registration.event.name
+            event_type_id = self.registration.event.type_id
+            username = self.first_name
+
+        text_options = signup_email_texts[notification_type]
+        texts = {
+            "heading": text_options["heading"] % {"username": username},
+            "text": text_options["text"][event_type_id] % {"event_name": event_name},
+        }
+
+        if notification_type == SignUpNotificationType.CANCELLATION:
+            texts["secondary_heading"] = text_options["secondary_heading"][
+                event_type_id
+            ] % {
+                "event_name": event_name,
+                "username": username,
+            }
+        elif notification_type == SignUpNotificationType.CONFIRMATION:
+            if self.signup_group:
```

Comment:
```suggestion
            if self.signup_group_id:
```
Evaluating the related object will make Django fetch the whole related object. Using only the related ID is preferred for performance benefits if only the ID is needed (https://docs.djangoproject.com/en/3.2/topics/db/optimization/#use-foreign-key-values-directly)